examples/python/api_demo/main.py

```diff
@@ -74,6 +73,36 @@ def run_segmentation() -> None:
     rr.log_text_entry("logs/seg_demo_log", "label1 disappears and everything with label3 is now default colored again")
 
 
+def transform_test() -> None:
+    rr.log_unknown_transform("transform_test/unknown", timeless=True)
+    rr.log_affine3(
+        "transform_test/child_from_parent_mat3",
+        child_from_parent=rr.TranslationMatrix3x3((123, 456, 789), np.array([[1, 2, 3], [4, 5, 6], [7, 8, 9]])),
+    )
+    rr.log_affine3("transform_test/parent_from_child_mat3", parent_from_child=rr.TranslationMatrix3x3((123, 456, 789)))
+    rr.log_affine3("transform_test/empty_translation_mat3", parent_from_child=rr.TranslationMatrix3x3())
+    rr.log_affine3("transform_test/translation_only", parent_from_child=rr.TranslationRotationScale3D((1, 2, 3)))
+    rr.log_affine3(
+        "transform_test/quaternion_only", parent_from_child=rr.TranslationRotationScale3D(rotation=(0, 0, 0, 1))
+    )
```

Comment:
This is just too much typing for a common use-case. The `parent_from_child`, while explicit, was already a mouthful and this pushes us well past an acceptable threshold.

Maybe we'd be better off with an approach that uses a bunch of keyword args and finds the best match as long as it's not inconsistent.
```
rr.log_child_transform("transform_test/mat33", translation=(123, 456, 789), mat33=rr.Identity)
rr.log_child_transform("transform_test/full", translation=(123, 456, 789), rot_axis=(0,1,0), degrees=45)
```

Reply:
Granted, maybe part of the problem is the example itself... people really *shouldn't* be using the APIs inlined like this. Something that looked more like:
```
def get_head_pose(...) -> rr.TranslationMatrix3x3:
  translation = #math
  rotation = #math
  return rr.TranslationMatrix3x3(translation, rotation)

...

rr.log_affine3("world/robot", parent_from_child=get_head_pose(...)
```
Might be more reasonable.

Reply:
Wondering how we can get best of both worlds. I think both the enforced dataclass way and the named-argument driven way should be possible. It's questionable if we want to put both on the same function.
On the direction part: I come from a world where there is only ever `parent_from_child` never anything else, so for me there is a clear default which could inform the API, but I suppose in robotics and computer vision that's not a given?

Reply:
If there's a clear default we can do this:
`log_affine3("world/robot", translation=..., rotation=..., scale=..., mat3=..., mat4=... child_from_parent=...)`
with every single argument optional and "type guessing" (mat args would exclude stuff ofc and mat4 is not implemented yet)